fe/fe-core/src/main/java/org/apache/doris/analysis/TableRef.java

```diff
@@ -499,6 +515,27 @@ protected void analyzeHints() throws AnalysisException {
         }
     }
 
+    public void analyzeTableSnapshot(Analyzer analyzer) throws AnalysisException {
+        if (tableSnapshot == null) {
+            return;
+        }
+        TableIf.TableType tableType = this.getTable().getType();
+        if (tableType != TableIf.TableType.HMS_EXTERNAL_TABLE) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_NONSUPPORT_TIME_TRAVEL_TABLE);
+        }
+        HMSExternalTable extTable = (HMSExternalTable) this.getTable();
+        if (extTable.getDlaType() != HMSExternalTable.DLAType.ICEBERG) {
```

Comment:
The dlaType may not be initialized, you can wrap this logic inside the `HMSExternalTable`, and call `makeSureInitialized()` before using `dlaType`